### PR TITLE
DRILL-7153: Drill Fails to Build using JDK 1.8.0_65

### PR DESCRIPTION
**Pull request by cgivre**
This PR fixes a bug in which building Drill using JDK 1.8.0_65 results in the following error. 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project drill-java-exec: Compilation failure
[ERROR] /Users/cgivre/github/drill-dev/drill/exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java:[59,68] error: unreported exception E; must be caught or declared to be thrown
[ERROR]   where E,T,V are type-variables:
[ERROR]     E extends Exception declared in method <T,V,E>accept(ExprVisitor<T,V,E>,V)
[ERROR]     T extends Object declared in method <T,V,E>accept(ExprVisitor<T,V,E>,V)
[ERROR]     V extends Object declared in method <T,V,E>accept(ExprVisitor<T,V,E>,V)
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :drill-java-exec
```

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java [hunk 1]
```diff
@@ -56,7 +56,8 @@ private FilterEvaluatorUtils() {
 
   public static RowsMatch evalFilter(LogicalExpression expr, MetadataBase.ParquetTableMetadataBase footer,
                                      int rowGroupIndex, OptionManager options, FragmentContext fragmentContext) {
-    List<SchemaPath> schemaPathsInExpr = new ArrayList<>(expr.accept(new FieldReferenceFinder(), null));
+    List<SchemaPath> schemaPathsInExpr = new ArrayList<>(expr.<Set<SchemaPath>,
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java]: @cgivre, as discussed in mail thread, could you please add a comment which explains why this change is required with the link to JDK issue which causes a compilation error.

Also, as discussed in mail thread, please add suppression to avoid warnings in IDE.

Comment (cgivre) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java]: I'm not sure which JDK issue references this specific problem, but I put a link to the one from the email chain.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java [hunk 2]
```diff
@@ -54,9 +54,12 @@
   private FilterEvaluatorUtils() {
   }
 
+  @SuppressWarnings("unchecked")
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java]: Please change it to `RedundantTypeArguments`, since current suppression does not help.
```suggestion
  @SuppressWarnings("RedundantTypeArguments")
```

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java [hunk 3]
```diff
@@ -54,9 +54,12 @@
...
   public static RowsMatch evalFilter(LogicalExpression expr, MetadataBase.ParquetTableMetadataBase footer,
                                      int rowGroupIndex, OptionManager options, FragmentContext fragmentContext) {
-    List<SchemaPath> schemaPathsInExpr = new ArrayList<>(expr.accept(new FieldReferenceFinder(), null));
+    //This change is necessary to prevent build errors on JDK 1.8.0_65.  See (https://bugs.openjdk.java.net/browse/JDK-8066974)
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java]: Please rephrase it to:
```suggestion
    // Specifies type arguments explicitly to avoid compilation error caused by JDK-8066974
```

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java [hunk 4]
```diff
@@ -54,9 +54,12 @@
...
+    List<SchemaPath> schemaPathsInExpr = new ArrayList<>(expr.<Set<SchemaPath>,
+            Void, RuntimeException>accept(new FieldReferenceFinder(), null));
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java]: Please change these two lines to specify all type arguments in the same line, for example like there:
```
List<SchemaPath> schemaPathsInExpr = new ArrayList<>(
        expr.<Set<SchemaPath>, Void, RuntimeException>accept(new FieldReferenceFinder(), null));
```

Comment (cgivre) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java]: Done and commits squashed.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java [hunk 5]
```diff
@@ -54,9 +54,12 @@
   private FilterEvaluatorUtils() {
   }
 
+  @SuppressWarnings("RedundantTypeArguments")
   public static RowsMatch evalFilter(LogicalExpression expr, MetadataBase.ParquetTableMetadataBase footer,
                                      int rowGroupIndex, OptionManager options, FragmentContext fragmentContext) {
-    List<SchemaPath> schemaPathsInExpr = new ArrayList<>(expr.accept(new FieldReferenceFinder(), null));
+    // Specifies type arguments explicitly to avoid compilation error caused by JDK-8066974
+    List<SchemaPath> schemaPathsInExpr = new ArrayList<>(
+            expr.<Set<SchemaPath>, Void, RuntimeException>accept(new FieldReferenceFinder(),null));
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/FilterEvaluatorUtils.java]: Please add space back:
```suggestion
            expr.<Set<SchemaPath>, Void, RuntimeException>accept(new FieldReferenceFinder(), null));
```
